### PR TITLE
scx_rustland: cpu topology refactoring

### PR DESCRIPTION
**Pull request by Decave**
The new topology crate allows us to replace the custom rustland topology
logic with the logic in the topology crate itself. Let's update rustland accordingly.

Note that this also requires us to add lookups from the root topology object to
all of the Cpu and Core objects in the system. No harm, no foul!

### CODE REVIEW

File: rust/scx_utils/src/topology.rs
Comment (htejun) [inline rust/scx_utils/src/topology.rs:203]: I think one approach we can take here is just restarting the scheduler after an hotplug event. I think some embedded applications use hoptlug for power management but outside of that it's a really really cold path, so it's unlikely to be worthwhile to support it in a fully dynamic fashion. We can e.g. provide a scx_utils sched trait where individual schedulers implement load / init callbacks and then scx_utils code can just unload and reload the scheduler whenever topology changes so that all schedulers get hotplug function without much effort.

**Review by htejun — APPROVED** [review]

File: rust/scx_utils/src/topology.rs
Comment (Decave) [reply rust/scx_utils/src/topology.rs:203]: Ahhh, great point, that does sound like a much better solution. Let me remove the mention of hotplug from the comment.

**Comment by Decave** [comment]
I'll let @arighi take a look before I land.

**Review by arighi — APPROVED** [review]
Looks great, thanks!

**Pull request merged**
